qiskit_experiments/library/characterization/analysis/ramsey_xy_analysis.py

```diff
@@ -108,45 +108,77 @@ def _generate_fit_guesses(
-
-        freq_val = float(np.average(freq_guesses))
-        user_opt.p0.set_if_empty(base=np.average(base_guesses))
-
-        # Guess the exponential decay by combining both curves
-        data_x = curve_data.get_subset_of("X")
-        data_y = curve_data.get_subset_of("Y")
-        decay_data = (data_x.y - user_opt.p0["base"]) ** 2 + (data_y.y - user_opt.p0["base"]) ** 2
-
+        # At very low frequency, y value of X (Y) curve stay at P=1.0 (0.5) for all x values.
+        # Computing y peak-to-peak with combined data gives fake amplitude of 0.25.
+        # Same for base, i.e. P=0.75 is often estimated in this case.
+        avg_y_ptp = 0.5 * (np.ptp(ramx_data.y) + np.ptp(ramy_data.y))
+        max_y = np.max(np.abs(curve_data.y))
+
+        if avg_y_ptp < 0.1 * max_y:
+            # When peak-to-peak value is less than 10% of full data range,
+            # likely frequency is too low to estimate.
+            # In this case, Ramsey Y curve corresponds to P=0.5, i.e. base.
+            avg_x = np.average(ramx_data.y)
+            avg_y = np.average(ramy_data.y)
+            user_opt.p0.set_if_empty(
+                amp=avg_x - avg_y,
+                tau=100 * np.max(curve_data.x),
+                base=avg_y,
+                phase=0.0,
+                freq=0.0,
+            )
+            return user_opt
+
+        base_guess = curve.guess.constant_sinusoidal_offset(curve_data.y)
         user_opt.p0.set_if_empty(
-            tau=-curve.guess.exp_decay(data_x.x, decay_data),
-            amp=0.5 * y_ptp,
+            amp=0.5 * avg_y_ptp,
+            base=base_guess,
             phase=0.0,
         )
 
-        opt_fp = user_opt.copy()
-        opt_fp.p0.set_if_empty(freq=freq_val)
-
-        opt_fm = user_opt.copy()
-        opt_fm.p0.set_if_empty(freq=-freq_val)
-
-        return [opt_fp, opt_fm]
+        # Guess the exponential decay by combining both curves
+        decay_data = (ramx_data.y - user_opt.p0["base"]) ** 2 + (
+            ramy_data.y - user_opt.p0["base"]
+        ) ** 2
+        if decay_data[0] - decay_data[-1] < 0.95 * 0.5 * avg_y_ptp:
+            # When decay is less than 95 % of peak-to-peak value, ignore decay and
+            # set large enough tau value compared with the measured x range.
+            user_opt.p0.set_if_empty(tau=1000 * np.max(curve_data.x))
+        else:
+            user_opt.p0.set_if_empty(tau=-1 / curve.guess.exp_decay(ramx_data.x, decay_data))
+
+        # Guess the oscillation frequency
+        xyfreq_guesses = []
+        for series in (ramx_data, ramy_data):
+            # Remove offset to eliminate DC peak
+            xyfreq_guesses.append(curve.guess.frequency(series.x, series.y - base_guess))
+        freq_val = float(np.average(xyfreq_guesses))
+
+        # FFT might be up to 1/2 bin off
+        df = 2 * np.pi / (np.min(np.diff(ramx_data.x)) * ramx_data.x.size)
+        freq_guesses = [freq_val - df, freq_val + df, freq_val]
```

Comment:
I'm not sure if the fitting to the spectrum gives really reasonable estimate. Because of the limited queue and delay value range, we sometimes cannot gain enough resolution around the FFT peak.
![image](https://user-images.githubusercontent.com/39517270/184817326-477564d9-3261-4919-a93a-26945b170f5c.png)